math_.py

```diff
@@ -13,3 +13,14 @@ def substract(a: int, b: int) -> int:
 def list_substract(a: List, b: List) -> List:
     assert len(a)==len(b), "len(a) != len(b)"
     return [i-j for i, j in zip(a,b)]
+
+def isnum(a: Any):
+    if isinstance(a, int):
+        return True
+    elif isinstance(a, str):
```

Comment:
The `isnum` function could be improved by using the `isdigit()` method for strings instead of trying to cast to an integer and catching exceptions. This makes the code more readable and efficient.